src/components/Questionnaire/EntitySelectionSheet.tsx

```diff
@@ -133,7 +134,7 @@ export function EntitySelectionSheet({
 
   return (
     <>
-      {system === "system-medication" ? (
+      {onProductEntitySelected ? (
```

Comment:
Let's not do this. Adding an extra prop is clearer in intent (plus, separation of concerns). While this works now, there's no guarantee that onProductEntitySelected will always be provided in future use cases.